src/zimdump.cpp

```diff
@@ -327,9 +306,12 @@ void ZimDumper::dumpFiles(const std::string& directory, bool symlinkdump)
 #endif
 
   std::set<char> ns;
-  for (zim::File::const_iterator it = pos; it != file.end(); ++it)
+  for (zim::File::const_iterator it = m_file.begin(); it != m_file.end(); ++it)
   {
     std::string d = directory + SEPARATOR + it->getNamespace();
+    if (nsfilter(it->getNamespace()))
```

Comment:
It is "funny", I would have written the other way (the filter must return `true` to keep the article). As for in python filter (https://docs.python.org/3/library/functions.html#filter) and c++ copy_if (https://en.cppreference.com/w/cpp/algorithm/copy)

But it works. No problem.

Reply:
Changed. I also think it makes a little bit more sense